### ISSUE TITLE
Issue with /docs/user-guide/node-selection/: optional "namespaces" field for podAntiAffinity doesn't work as documented

### ISSUE THREAD

**Issue by curtis628** [kubernetes/kubernetes.github.io#2676]
**This is a...** 

- [ ] Feature Request
- [x] Bug Report

**Problem:**
The optional `namespaces` field for `podAntiAffinity` doesn't work according to the [the docs page for anti-affinity](https://kubernetes.io/docs/user-guide/node-selection/#inter-pod-affinity-and-anti-affinity-alpha-feature) which says

> If [`namespaces`] is omitted, it defaults to the namespace of the pod where the affinity/anti-affinity definition appears. If defined but empty, it means “all namespaces.”

I discovered that if `namespaces` is omitted, it defaults to "all namespaces" 

**Proposed Solution:**
Change the docs or change the default behavior to match the docs

**Page to Update:**
https://kubernetes.io/docs/user-guide/node-selection/#inter-pod-affinity-and-anti-affinity-alpha-feature

Kubernetes Version: 1.5.2
Hosted on: AWS
Number of minions: 17 (although likely not a requirement)

**Steps to Reproduce**
1. Create a k8s namespace `test-ns1`: `kubectl create ns test-ns1`
2. Create a test deployment YAML, with anti-affinity, and N-1 replicas (where N is number of k8s nodes). Leave out the optional `namespaces` field in the annotation to get the default behavior.

Save .yaml file below to: `./nginx-deployment.yaml`

```yaml
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: nginx-deployment
spec:
  replicas: 16
  template:
    metadata:
      labels:
        app: nginx
      annotations:
        scheduler.alpha.kubernetes.io/affinity: >
          {
            "podAntiAffinity": {
              "requiredDuringSchedulingIgnoredDuringExecution": [
                {
                  "labelSelector": {
                    "matchExpressions": [
                      {
                        "key": "app",
                        "operator": "In",
                        "values": ["nginx"]
                      }
                    ]
                  },
                  "topologyKey": "kubernetes.io/hostname"
                }
              ]
            }
          }
    spec:
      containers:
      - name: nginx
        image: nginx:1.7.9
        ports:
        - containerPort: 80
```
3. Create deployment on `test-ns1`: `kubectl --namespace=test-ns1 create -f ./nginx-deployment.yaml --record`

> Confirm all pods are scheduled (and on different nodes)

4. Create a separate ns: `kubectl create ns test-ns2`

5. Create deployment on `test-ns2`: `kubectl --namespace=test-ns2 create -f ./nginx-deployment.yaml --record`

_Expected Behavior_:
All pods should be able to placed since this is a separate namespace. The YAML didn't specify `namespaces` ... so I expected the default behavior to only use anti-affinity pods within the namespace and not across _all_ namespaces.

_Actual Behavior_:
Some `test-ns2` pods remaining in a "Pending" state. Doing a `kubectl --namespace test-ns2 describe pod [pod-name]` on a "Pending" pod shows the anti-affinity is preventing the pods from being replaced even though it's in a separate namespace.

```
Events:
  FirstSeen	LastSeen	Count	From			SubobjectPath	Type		Reason			Message
  ---------	--------	-----	----			-------------	--------	------			-------
  2m		20s		14	{default-scheduler }			Warning		FailedScheduling	pod (nginx-deployment-2310940957-gtdgv) failed to fit in any node
fit failure summary on nodes : MatchInterPodAffinity (17)
```

NOTE: If I add a `namespaces` field with a value in the anti-affinity annotation, then the affinity is only applied on the namespace specified... but I expected that to be the default behavior (leaving `namespaces` out of the `podAntiAffinity` annotation altogether).

**Comment by erikgrinaker** [comment]
I think this is a bug in the implementation, not the docs. The Kubernetes source code seems to confirm this:

https://github.com/kubernetes/kubernetes/blob/e443277d58e3ff1f0701ab97a1f4932389c27847/pkg/api/types.go#L1904-L1906

Perhaps resubmit this as an issue on the main Kubernetes repo?

**Comment by erikgrinaker** [comment]
Just an update: this is indeed a bug in the implementation, and has been fixed on the 1.7 branch. A backport to 1.6 is being discussed in https://github.com/kubernetes/kubernetes/issues/45484.

**Comment by curtis628** [comment]
@erikgrinaker - Thanks for submitting this on the main k8s repo, and keeping me posted! Closing this doc bug.
